pallets/staking_rewards/src/lib.rs

```diff
@@ -208,6 +236,14 @@ impl<T: Config> EraPayout<BalanceOf<T>> for Module<T> {
         total_issuance: BalanceOf<T>,
         era_duration_millis: u64,
     ) -> (BalanceOf<T>, BalanceOf<T>) {
+        if Self::high_rate_rewards_end_at()
```

Comment:
The correct solution must be to declare the reward duration as the number of eras and start by the following sequential era after the runtime upgrade so that the period will be fair.
In the case you described even 1 block will result in high rewards paid at the end of the era.

Reply:
Yes, I was erring on the generous side.